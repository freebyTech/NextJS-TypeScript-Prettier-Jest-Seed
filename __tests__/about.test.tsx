import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";

import App from "../pages/about";

describe("With Enzyme", () => {
  it('App shows "About Next.js"', () => {
    const app = shallow(<App />);

    expect(app.find("p").text()).toEqual("About Next.js");
  });
});

describe("With Snapshot Testing", () => {
  it('App shows "About Next.js"', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
