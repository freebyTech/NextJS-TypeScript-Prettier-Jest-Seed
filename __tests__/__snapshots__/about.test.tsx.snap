// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "About Next.js" 1`] = `
<div
  style={
    Object {
      "border": "1px solid #DDD",
      "margin": 20,
      "padding": 20,
    }
  }
>
  <div>
    <a
      href="/"
      onClick={[Function]}
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      Home
    </a>
    <a
      href="/about"
      onClick={[Function]}
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      About
    </a>
  </div>
  <p>
    About Next.js
  </p>
</div>
`;
